packages/editor/README.md

```diff
@@ -370,14 +370,328 @@ function Toolbar() {
+
+However, most core behaviors and most of your custom behaviors are most likely going to be conditional since you don't want them to run all the time.
+
+An example of a conditional behavior that can be found in the core is the "clear list on Enter" behavior:
+
+```ts
+const clearListOnEnter = defineBehavior({
+  on: 'insert.break',
+  guard: ({context}) => {
+    const selectionCollapsed = selectionIsCollapsed({context})
+    const focusListBlock = getFocusListBlock({context})
+
+    if (
+      !selectionCollapsed ||
+      !focusListBlock ||
+      !isEmptyTextBlock(focusListBlock.node)
+    ) {
+      return false
+    }
+
+    return {focusListBlock}
+  },
+  actions: [
+    (_, {focusListBlock}) => [
+      {
+        type: 'text block.unset',
+        props: ['listItem', 'level'],
+        at: focusListBlock.path,
+      },
+    ],
+  ],
+})
+```
+
+This time we listen for the `insert.break` event (<kbd>Enter</kbd>) and here, the `guard` checks
+
+1. that the selection is collapsed,
+2. that the currently focused block is a list block
+3. and that this list block is empty
+
+If either of these conditions are falsy, the `guard` returns `false` and the actions won't fire. This allows the `insert.break` event to propagate through the system. If no other behavior overwrites it, the default action is triggered and the block is split as usual.
+
+However, if all conditions are truthy, the `guard` returns an object with the `focusListBlock` (a truthy value) which then fires the `text block.unset` action to unset the `listItem` and `level` props on the block, effectively turning it into an ordinary text block.
+
+The guard is able to determine this because it has access to the current editor `context`. The `EditorContext` includes the editor `schema`, the current `selection` and the current `value`:
+
+```ts
+type EditorContext = {
+  schema: EditorSchema
+  selection: EditorSelection
+  value: Array<PortableTextBlock>
+}
+```
+
+`selectionIsCollapsed` and `getFocusListBlock` are just pure functions (called "selectors") that derive state from this context. Both are built-in and exported from `@portabletext/editor/selectors` (along with a number of other useful, built-in selectors).
+
+```ts
+import {
+  getFocusListBlock,
+  selectionIsCollapsed,
+} from '@portabletext/editor/selectors'
+```
+
+And you can even write your own selectors:
+
+```ts
+import type {EditorSelector} from '@portabletext/editor/selectors'
+
+const selectionIsBackwards: EditorSelector<boolean> = ({context}) => {
+  // ...
+}
+```
+
+This behavior also shows another neat trick:
+
+> 💡 You can return **parameters** from the `guard` that you can reuse when firing actions
+
+This is extremely useful since you often want to perform a set of actions usings state selected in the guard. Instead of having to pluck that state again in the action callback, you can just return and reuse it.
```

Comment:
```suggestion
This is extremely useful since you often want to perform a set of actions using state selected in the guard. Instead of having to pluck that state again in the action callback, you can just return and reuse it.
```